app/code/core/Mage/Catalog/Model/Product/Indexer/Flat.php

```diff
@@ -151,7 +151,7 @@ public function matchEvent(Mage_Index_Model_Event $event)
             } else {
                 /** @var Mage_Core_Model_Store $store */
                 $store = $event->getDataObject();
-                if ($store && $store->isObjectNew()) {
+                if ($store && ($store->isObjectNew() || $store->getOrigData('is_active') != $store->getIsActive() && $store->getIsActive())) {
```

Comment:
maybe it would be best to wrap the new condition into a new statement?

Reply:
I'm wondering if we need the `$store->getOrigData('is_active') != $store->getIsActive()` part, if the store is active `$result` must be `true` anyway? am I missing something?

Reply:
Yes, I thought that too, but I used the same code I found here which catches the EAV reindex alert too:



app/code/core/Mage/Catalog/Model/Product/Indexer/Eav.php::_registerEvent()@123
```
protected function _registerEvent(Mage_Index_Model_Event $event)
    {
        $entity = $event->getEntity();

        if ($entity == Mage_Catalog_Model_Product::ENTITY) {
            switch ($event->getType()) {
                case Mage_Index_Model_Event::TYPE_DELETE:
                    $this->_registerCatalogProductDeleteEvent($event);
                    break;

                case Mage_Index_Model_Event::TYPE_SAVE:
                    $this->_registerCatalogProductSaveEvent($event);
                    break;

                case Mage_Index_Model_Event::TYPE_MASS_ACTION:
                    $this->_registerCatalogProductMassActionEvent($event);
                    break;
            }
        } elseif ($entity == Mage_Catalog_Model_Resource_Eav_Attribute::ENTITY) {
            switch ($event->getType()) {
                case Mage_Index_Model_Event::TYPE_SAVE:
                    $this->_registerCatalogAttributeSaveEvent($event);
                    break;
            }
        } elseif ($entity == Mage_Catalog_Model_Convert_Adapter_Product::ENTITY) {
            $event->addNewData('catalog_product_eav_reindex_all', true);
        } elseif ($entity == Mage_Core_Model_Store::ENTITY) {
            if ($event->getType() === Mage_Index_Model_Event::TYPE_SAVE) {
                /** @var Mage_Core_Model_Store $store */
                $store = $event->getDataObject();
                if ($store->getOrigData('is_active') != $store->getIsActive() && $store->getIsActive()) {
                    $event->getProcess()->changeStatus(Mage_Index_Model_Process::STATUS_REQUIRE_REINDEX);
                }
            }
        }
    }
```


I also tried to recreate a new store view, and it seems to work correctly.

Reply:
if think this means

if is a new store view => true
if is not a new store view and is_active is changed and is changed in an active status => true

Then, if you save with an inactive status, the alert will not be shown.